### DISCUSSION TITLE
When is defaultQuery applied

### DISCUSSION THREAD

**Discussion by thedangler** [saloonphp/saloon#100] [Q&A] [answered]
I'm trying to set some default query data for my post request, however it looks like its never applied at all.

I've tried not passing in any data to be merged with the default query params and it's always blank.

       //$queryVariables  comes from command line page_limit=20

        $request = new QueryRequest(
            $this->argument('client_key'),
            $queryVariables
        );

// defaultQuery in Request

        return [
            'security_key' => $this->security_key,
            'result_limit' => 50,
            'page_number' => 0,
            'start_date' => $startDate,
            'end_date' =>  $endDate,
            'condition' => 'complete',
        ];

I've tried merging the queries, not setting them in the command, yet the default queries never get applied.

What am I missing?

**Comment by Sammyjo20** [comment, marked as answer]
Hey @thedangler are you able to share your full request class please?

The defaultQuery is merged in inside of the RequestManager class when you send your request, so it won't be visible if you were to view the request's query parameters before it is sent.
